core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -76,6 +77,21 @@ class BrokerMetadataListener(
    */
   private var _highestTimestamp = -1L
 
+  /**
+   * The log epoch of the current transaction, or -1 if there is none.
+   */
+  private var _transactionEpoch = -1;
+
+  /**
+   * The log offset of the current transaction, or -1L if there is none.
+   */
+  private var _transactionOffset = -1L;
+
+  /**
+   * Pending records in the current transaction, or null if there is none such.
+   */
+  private var _transactionRecords: util.ArrayList[ApiMessageAndVersion] = _
+
```

Comment:
I'm guessing all of this logic will go away once we refactor the broker to use MetadataLoader?

Reply:
yeah, this file will be removed. but that's too ambitious for 3.4 I think